src/components/getApplicationData.ts

```diff
@@ -34,10 +37,12 @@ export const getApplicationData = async (payload: ActionPayload) => {
   }
 
   const reviewData =
-    trigger_payload.table === 'review'
+    input?.reviewId || trigger_payload?.table === 'review'
       ? {
-          reviewId: trigger_payload.record_id,
-          ...(await DBConnect.getReviewData(trigger_payload.record_id)),
+          reviewId: input?.reviewId ?? trigger_payload?.record_id,
+          ...(await DBConnect.getReviewData(
+            input?.reviewId ?? (trigger_payload?.record_id as number)
+          )),
         }
       : {}
```

Comment:
This is a hard to read as it is, I would suggest replace with:
```
const isReview = input?.reviewId ||  trigger_payload?.table === 'review'
const review = isReview ? await DBConnect.getReviewData(
            input?.reviewId ?? (trigger_payload?.record_id as number)
          ) : undefined

const reviewData = isReview ? {
      reviewId: input?.reviewId ?? trigger_payload?.record_id,
          ...{review},
     } : {}
```

Reply:
Okay, I did something slightly different which I think is much clearer. b11f088f